static/app/views/dashboardsV2/manage/index.tsx

```diff
@@ -220,6 +222,18 @@ class ManageDashboards extends AsyncView<Props, State> {
     });
   }
 
+  onAdd(dashboard: DashboardDetails) {
+    const {organization, api} = this.props;
+    trackAdvancedAnalyticsEvent('dashboards_manage.templates.add', {
+      organization,
+      dashboard_id: dashboard.id,
+    });
+
+    createDashboard(api, organization.slug, dashboard, true).then(() =>
+      this.onDashboardsChange()
+    );
```

Comment:
why not make this function async, so we can


```suggestion
    await createDashboard(api, organization.slug, dashboard, true)
    this.onDashboardsChange()
```